src/wallet/spend.cpp

```diff
@@ -1161,7 +1161,19 @@ static util::Result<CreatedTransactionResult> CreateTransactionInternal(
     if (!select_coins_res) {
         // 'SelectCoins' either returns a specific error message or, if empty, means a general "Insufficient funds".
         const bilingual_str& err = util::ErrorString(select_coins_res);
-        return util::Error{err.empty() ?_("Insufficient funds") : err};
+        if (!err.empty()) return util::Error{err};
+
+        // Check if we have enough balance but cannot cover the fees
+        CAmount available_balance = preset_inputs.total_amount + available_coins.GetTotalAmount();
+        if (available_balance >= recipients_sum) {
+            CAmount available_effective_balance = preset_inputs.total_amount + available_coins.GetEffectiveTotalAmount().value_or(available_coins.GetTotalAmount());
```

Comment:
I’m wondering whether there is a bug here. It seems to me that this is assuming that the effective value is only available when SFFO is not being used, but `COutput.effective_value` is set and aggregated whenever the COutputs have a feerate. Does this not explicitly need to handle SFFO? I.e., maybe this needs to be something along the lines of:

```suggestion
            CAmount available_coins_total_amount = coin_selection_params.m_subtract_fee_outputs ? available_coins.GetTotalAmount() : available_coins.GetEffectiveTotalAmount();
            CAmount available_effective_balance = preset_inputs.total_amount +available_coins_total_amount;
```

Reply:
Since fees are not included when SFFO, I don't think this is reachable as should `available_balance` is always greater than `recipients_sum` in that case, so I don't think this needs any special SFFO treatment. Furthermore, if it were possible to reach this with SFFO, I think the error message would be confusing and it would actually be better to return "Insufficent Funds", so this whole check could probably be guarded by an `if (m_subtract_fee_outputs)` to skip it when SFFO is on.

Reply:
ha.. true. It seems I'm not putting much brain power here.. updated per feedback. Thanks.

Reply:
re: https://github.com/bitcoin/bitcoin/pull/25269#discussion_r2115800834

Thinking about this more I can see why SFFO should make the "amount exceeds your balance when fees are included" condition harder to trigger than when SFFO is not used, because when SFFO is enabled the wallet is able to reduce the amounts recipients receive to cover any fees, whereas without SFFO the amount received needs to be fixed. But I'd still expect the `(available_balance >= recipients_sum && available_effective_balance < selection_target)` condition to be able to trigger when SFFO is used if there are coins are which are not economical to send. Also even if there is other code making this condition difficult or impossible to trigger when SFFO is used, I still don't see a reason this code should be checking for SFFO specially. Even with SFFO, an error message that mentions fees and includes the attempted fee would still be more descriptive and helpful than the generic "insufficient funds" message.

Reply:
re: https://github.com/bitcoin/bitcoin/pull/25269#discussion_r2115800834

Update: think I finally figured this out. In the SFFO case the fee exceeds balance error can still happen but the `available_balance >= recipients_sum` condition will not detect it. So I think the change I would suggest would just be dropping the `m_subtract_fee_outputs` code and writing a more descriptive comment:

```diff
--- a/src/wallet/spend.cpp
+++ b/src/wallet/spend.cpp
@@ -1164,12 +1164,19 @@ static util::Result<CreatedTransactionResult> CreateTransactionInternal(
 
         // Check if we have enough balance but cannot cover the fees
         CAmount available_balance = preset_inputs.total_amount + available_coins.GetTotalAmount();
-        // Note: Since SFFO reduces existing outputs to cover fees, recipients_sum should already be
-        // strictly less than available_balance when enabled. Still, as a sanity-check, skip when SFFO is enabled.
-        if (available_balance >= recipients_sum && !coin_selection_params.m_subtract_fee_outputs) {
+        if (available_balance >= recipients_sum) {
             CAmount available_effective_balance = preset_inputs.total_amount + *available_coins.GetEffectiveTotalAmount();
             if (available_effective_balance < selection_target) {
-                return util::Error{_("The total transaction amount exceeds your balance when fees are included")};
+                // Note: this fees exceeds balance error will not currently
+                // trigger when SFFO (m_subtract_fee_outputs) is enabled because
+                // the effective balance will equal the original balance in this
+                // case. This is not ideal because it fails to provide the most
+                // descriptive error message, but it's not worth making this
+                // code more complex by checking the sizes of outputs fees can
+                // be subtracted from. Additionally, fees exceeding the balance
+                // should happen more rarely in the SFFO case given the ability
+                // to take fees from outputs.
+                return util::Error{strprintf(_("The total exceeds your balance when the %s transaction fee is included."), FormatMoney(selection_target - recipients_sum))};
             }
         }
 
```